packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts

```diff
@@ -384,4 +385,79 @@ export class DependencyManager {
         );
     }
   }
+
+  /**
+   * Converts an package name of a pre-built provider package in target language to the name in npm
+   * Inverse of: `convertPackageName`
+   */
+  private convertFromPackageNameToNpm(name: string): string {
+    const npmPackagePrefix = "@cdktf/provider-";
+    const regexes = {
+      [Language.GO]: /github.com\/(cdktf|hashicorp)\/cdktf-provider-(.+)-go\//i,
+      [Language.TYPESCRIPT]: /(.+)/i,
+      [Language.CSHARP]: /HashiCorp\.Cdktf\.Provider\.(.+)/i,
+      [Language.JAVA]: /com\.hashicorp\.cdktf-provider-(.+)/i,
+      [Language.PYTHON]: /cdktf-cdktf-provider-(.+)/i,
+    };
+    const regex = regexes[this.targetLanguage];
+    if (!regex) {
+      throw new Error("Language not supported for pre-built providers");
```

Comment:
I think we can use one of our [custom errors](https://github.com/hashicorp/terraform-cdk/blob/86e8afa3d35fdcfeeca73ba6a70872b1fb06a8b1/packages/cdktf-cli/src/lib/errors.ts#L40) here instead

Reply:
Nice! Didn't know about those.